### PR TITLE
Rubocop rules extension 3

### PR DESCRIPTION
**Pull request by danielkryska**
For the review, some of the cop changes are split into separate commits.
Most of them (without prominent logic changes) will be squashed before merging.

-------------------------

Some Cops enabling commits are squashed together. However, some remain expanded to be easier to analyse in the future.

### CODE REVIEW

File: app/helpers/image_helper.rb [hunk 1]
```diff
@@ -3,18 +3,10 @@
...
-    @@default_logo_path
-  end
-
-  def self.permitted_ext_message
-    @@permitted_ext_message
-  end
+  DEFAULT_LOGO_PATH = Webpacker.manifest.lookup("media/images/eosc-img.png")
 
   def self.to_temp_file(logo, ext)
     tmp_logo = Tempfile.new(["logo_temp", ext])
```
Comment (Marcelinna) [inline app/helpers/image_helper.rb:12]: changes in  app/helpers/image_helper.rb looks fine? 
On PR is a problem with the image, so it is hard to check it.

Comment (jswk) [reply app/helpers/image_helper.rb:12]: Yes, they're fine. There were several affected tests in `spec/lib/import/resources_spec.rb`, which traced the modified code paths and are now corrected.

File: app/controllers/favourites/services_controller.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ def update
   private
 
   def many?
-    UserService.where(user: current_user).size > 1 || cookies[:favourites]&.size > 1
+    UserService.where(user: current_user).size > 1 || (cookies[:favourites]&.size || 0) > 1
```
Comment (kmarszalek) [inline app/controllers/favourites/services_controller.rb:25]: brackets

File: app/helpers/scientific_domains_helper.rb [hunk 3]
```diff
@@ -18,11 +18,11 @@ def grouped_scientific_domains
   private
 
   def group_scientific_domains!(result, current_path, key, values)
-    if !values.empty?
```
Comment (kmarszalek) [inline app/helpers/scientific_domains_helper.rb:21]: unless [']

**Review by kmarszalek — CHANGES_REQUESTED** [review]
git. git pl
squash smth, after Marcelinas approval and add brackets.

**Review by Marcelinna — APPROVED** [review]

**Review by kmarszalek — APPROVED** [review]

**Pull request merged**